### PR TITLE
test: fix TestGetMergeRegionSizeAndCount test

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/52095 https://github.com/pingcap/tidb/issues/52094

Problem Summary:
the backoff case will cost too mush time.
### What changed and how does it work?
use failpoint and return parse error immediately.
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/conn/conn_test.go [hunk 1]
```diff
@@ -406,6 +406,8 @@ func TestGetMergeRegionSizeAndCount(t *testing.T) {
 	}
 
 	ctx := context.Background()
+	failpoint.Enable("github.com/pingcap/tidb/br/pkg/conn/stop-retry-on-get-config-from-tikv", "return(true)")
```
Comment (hawkingrei) [inline br/pkg/conn/conn_test.go:409]: It should check whether to return the no error. it will return an error if failpoint is deleted.

**Review by Leavrth — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**